web/src/components/theme-provider.tsx

```
@@ -0,0 +1,50 @@
+  }
+  
+  // After initial render, enable theme switching
+  React.useEffect(() => {
+    setMounted(true)
+  }, [])
+  
+  return (
+    <>
+      {/* Add a script to prevent flash of wrong theme */}
+      {!mounted && (
+        <style dangerouslySetInnerHTML={{ __html: `
```

Biome (1.9.4)

[error] Avoid passing content using the dangerouslySetInnerHTML prop.

Setting content using code can expose users to cross-site scripting (XSS) attacks

(lint/security/noDangerouslySetInnerHtml)